psycopg/psycopg/postgres.py

```diff
@@ -108,18 +108,22 @@
 def register_default_adapters(context: AdaptContext) -> None:
 
     from .types import array, bool, composite, datetime, enum, json, multirange
-    from .types import net, none, numeric, range, string, uuid
+    from .types import net, none, numeric, numpy, range, string, uuid
 
     array.register_default_adapters(context)
-    bool.register_default_adapters(context)
     composite.register_default_adapters(context)
     datetime.register_default_adapters(context)
     enum.register_default_adapters(context)
     json.register_default_adapters(context)
     multirange.register_default_adapters(context)
     net.register_default_adapters(context)
     none.register_default_adapters(context)
-    numeric.register_default_adapters(context)
     range.register_default_adapters(context)
     string.register_default_adapters(context)
     uuid.register_default_adapters(context)
+
+    # Make sure to register dumpers to the standard types last, in order to
+    # pick them by default when looking up by oid.
+    numpy.register_default_adapters(context)
```

Comment:
Strange comment: numpy's types are not "standard types", no?

Reply:
Maybe this comment needs rewording then 🙂

What I meant is that there was [this failure](https://github.com/psycopg/psycopg/actions/runs/3716378652/jobs/6302634045#step:9:599). This was caused by the fact that both Python's Decimal and numpy's uint64 have a dumper registered with the Postgres' numeric oid. But the former covers the entire numeric domain, the latter just the integers. If the latter is registered later then, if we look up for dumpers by oids, we will pick it even in order to dump a Decimal, [we will cast it to int](https://github.com/vertefra/psycopg/blob/0807d176e1c01c5fd2a31363070b1c8b1c59f355/psycopg/psycopg/types/numpy.py#L67), dropping the decimal part.

The situation happens whenever there is more than one dumper registered which can produce the same oid. So far, IIRC, it only happened for [the dbapi-compliant Binary wrapper](https://github.com/psycopg/psycopg/blob/14bfb96c4ad53a0c426406a587bea5d45fd605e4/psycopg/psycopg/dbapi20.py#L110-L112), whose dumper is chosen in preference of the basic binary dumper, because it can cover both types of objects.